Debounce digitransit autocomplete requests

mapehe opened the pull request:
#### Summary

Avoid spamming digitransit API with autocomplete requests on each keystroke by introducing a timeout between the requests.

reynders reviewed (approved):
Tested locally as ok

mapehe commented:
Thanks for the comment @reynders. @Joosakur pointed out that `useDebounce` would probably belong to `lib-commons`. Added a small refactoring commit where I moved it there.
